chat_shell/start.sh

```
@@ -26,6 +26,35 @@ YELLOW='\033[1;33m'
+        if [ -n "$ip" ]; then
+            echo "$ip"
+            return
+        fi
+    fi
+
+    # Fallback to localhost
+    echo "localhost"
+}
+
+# Get local IP for backend URL (used by executor containers in Docker)
+export BACKEND_API_URL=$(get_local_ip)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)